src/rgen/Microsoft.Macios.Transformer/DataModel/Binding.Transformer.cs

```diff
@@ -234,6 +250,17 @@ internal Binding (InterfaceDeclarationSyntax interfaceDeclarationSyntax, INamedT
 		// loop over the different members and add them to the model
 		GetMembers<PropertyDeclarationSyntax, Property> (interfaceDeclarationSyntax, context, Skip, Property.TryCreate,
 			out properties);
+		// methods are a little diff, in the old SDK style, both methods and constructors are methods, we will get
+		// all exported methods and then filter accordingly
+		GetMembers<MethodDeclarationSyntax, Method> (interfaceDeclarationSyntax, context, Skip, Method.TryCreate,
+			out ImmutableArray<Method> allMethods);
+		methods = [.. allMethods.Where (m => !m.IsConstructor)];
+		constructors = allMethods.Where (m => m.IsConstructor)
+			.Select (m => m.ToConstructor ())
+			.Where (c => c is not null)
+			.Select (c => c!.Value)
+			.ToImmutableArray ();
```

Comment:
3 Wheres, 2 Selects. A LINQ query that could have been a for loop (for perf reasons). but the meaning is quite clear...